src/TypeScript/Rules/RuleDefinitionsProvider.cs

```diff
@@ -69,12 +58,38 @@ public RuleDefinitionsProvider([Import(RuleDefinitionsFilePathContractName)] str
 
         IEnumerable<RuleDefinition> IJavaScriptRuleDefinitionsProvider.GetDefinitions() => jsRules;
 
+        string ITypeScriptRuleKeyMapper.GetSonarRuleKey(string eslintRuleKey) =>
+            GetRuleKey(eslintRuleKey, tsRules);
+
+        string IJavaScriptRuleKeyMapper.GetSonarRuleKey(string eslintRuleKey) =>
+            GetRuleKey(eslintRuleKey, jsRules);
+
         private static List<RuleDefinition> Load(string filePath) =>
             JsonConvert.DeserializeObject<List<RuleDefinition>>(File.ReadAllText(filePath, Encoding.UTF8));
 
         private static IReadOnlyCollection<RuleDefinition> FilterByRepo(string repoPrefix, IEnumerable<RuleDefinition> rules) =>
             rules.Where(x => x.RuleKey.StartsWith(repoPrefix, System.StringComparison.OrdinalIgnoreCase))
             .ToList()
             .AsReadOnly();
+
+        private static string GetRuleKey(string eslintRuleKey, IEnumerable<RuleDefinition> rules)
+        {
+            var repoAndKey = rules.FirstOrDefault(x => eslintRuleKey.Equals(x.EslintKey, System.StringComparison.OrdinalIgnoreCase))
+                ?.RuleKey;
+            return repoAndKey == null ? null : StripRepoPrefix(repoAndKey);
+        }
+
+        private static string StripRepoPrefix(string ruleKey)
+        {
+            // Expecting the key to be in the form [repo id]:[rule id]
```

Comment:
I decided to strip the repo prefix so we'll display e.g. `S123` in the Error List rather than `javascript:S123`. It felt more in keeping with what VS developers would expect.

In contrast, the other SL flavours don't strip the repo prefix, but then they don't display the error codes in the same way either.

Reply:
It's not what we currently do for taint/hotspots/cfamily but I agree that it's neater. Do you think it's worth opening an issue(s) to align the other 3?